### PR TITLE
[CST] [Backend] Expanded status logging for claims

### PR DESCRIPTION
**Pull request by samcoforma**
## Summary

- Added the following to claim type logging in `benefits_claims_controller.rb`:
  - `attributes.claimPhaseDates.currentPhaseBack`
  - `attributes.claimPhaseDates.latestPhaseType`
  - `attributes.decisionLetterSent`
  - `attributes.developmentLetterSent`
- Updated `benefits_claims_controller_spec.rb`

## Related issue(s)

- [[CST] Expand status logging for claims](https://github.com/department-of-veterans-affairs/va.gov-team/issues/84632)

## Testing done

- [x] *New code is covered by unit tests*

## Acceptance criteria

- [x] We are logging the attribute that tells us if the claim is currently phased back or not
- [x] We are logging the attribute related to the current claim status
- [x] We are logging the attribute related to whether a decision letter has been sent
- [x] We are logging the attribute related to whether a development letter has been sent

### CODE REVIEW

File: app/controllers/v0/benefits_claims_controller.rb [hunk 1]
```diff
@@ -33,6 +33,10 @@ def show
                             claim_type_code: claim_info['claimTypeCode'],
                             num_contentions: claim_info['contentions'].count,
                             ep_code: claim_info['endProductCode'],
+                            current_phaseback: claim_info['claimPhaseDates']['currentPhaseBack'],
```
Comment (pmclaren19) [inline app/controllers/v0/benefits_claims_controller.rb:36]: I think this is supposed to be `current_phase_back`

File: spec/controllers/v0/benefits_claims_controller_spec.rb [hunk 2]
```diff
@@ -143,6 +143,10 @@
                   claim_type_code: '020NEW',
                   num_contentions: 1,
                   ep_code: '020',
+                  current_phaseback: false,
```
Comment (pmclaren19) [inline spec/controllers/v0/benefits_claims_controller_spec.rb:146]: I think this is supposed to be `current_phase_back`

File: spec/controllers/v0/benefits_claims_controller_spec.rb [hunk 3]
```diff
@@ -143,6 +143,10 @@
                   claim_type_code: '020NEW',
```
Comment (pmclaren19) [inline spec/controllers/v0/benefits_claims_controller_spec.rb:143]: @samcoforma what are you getting these values from? Im not seeing that you had to update any mock data that you changed here

Comment (samcoforma) [reply spec/controllers/v0/benefits_claims_controller_spec.rb:143]: @pmclaren19 from `spec/support/vcr_cassettes/lighthouse/benefits_claims/show/200_response.yml`

Comment (samcoforma) [reply spec/controllers/v0/benefits_claims_controller_spec.rb:143]: I didn't need to change the mock data, I just needed to start logging what was already there...

Comment (pmclaren19) [reply spec/controllers/v0/benefits_claims_controller_spec.rb:143]: Okay that seems to match up!

**Review by pmclaren19 — APPROVED** [review]

**Pull request merged**